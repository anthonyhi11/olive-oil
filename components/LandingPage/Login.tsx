import { useState } from "react";
import { Button, Form } from "react-bootstrap";
import Modal from "react-bootstrap/Modal";
import { store } from "../../reduxTest";

interface ILoginProps {
	show: boolean;
	handleClose: (arg1: boolean) => void;
}

function Login({ show, handleClose }: ILoginProps) {
	const [userName, setUserName] = useState<string>("");

	function doLogin() {
		handleClose(false);
		store.dispatch({ type: "loggedIn", payload: userName });
	}

	return (
		<>
			<Modal show={show} onHide={() => handleClose(false)}>
				<Modal.Header closeButton>
					<Modal.Title>Login</Modal.Title>
				</Modal.Header>
				<Modal.Body>
					<Form>
						<Form.Group className="mb-3" controlId="formBasicEmail">
							<Form.Label>Email address</Form.Label>
							<Form.Control
								type="email"
								placeholder="Enter email"
								onChange={(e) => setUserName(e.target.value)}
							/>
						</Form.Group>

						<Form.Group className="mb-3" controlId="formBasicPassword">
							<Form.Label>Password</Form.Label>
							<Form.Control type="password" placeholder="Password" />
						</Form.Group>
					</Form>
				</Modal.Body>
				<Modal.Footer>
					<Button variant="secondary" onClick={() => handleClose(false)}>
						Close
					</Button>
					<Button variant="primary" onClick={doLogin}>
						Login
					</Button>
				</Modal.Footer>
			</Modal>
		</>
	);
}

export default Login;
