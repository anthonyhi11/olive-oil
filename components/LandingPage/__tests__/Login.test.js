import React from "react";
import {render} from "@testing-library/react";
import Login from "../Login";

it("Renders and matches snapshot", () => {
	const {container} = render(<Login show={true} handleClose={() => {}} />);
	expect(container).toMatchSnapshot();
});
