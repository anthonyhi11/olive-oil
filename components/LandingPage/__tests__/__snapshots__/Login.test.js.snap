// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders and matches snapshot 1`] = `
<div
  aria-hidden="true"
/>
`;
